Regression: msbuild.exe throws NullReferenceException when /t:target1,target2 specified

When the `/t:` switch is used with more than one target specified (comma-delimited list), the latest msbuild.exe 16.4 version crashes instead of building both targets. Same crash if `/t:` is simply specified twice, once for each target.

MSBuild version: 16.4.0.50805

## Repro steps
```
md repro
cd repro
dotnet new classlib
dotnet new sln
dotnet sln add repro.csproj
msbuild.exe /restore /t:build,pack
```

## Output

```
  Restore completed in 47.17 ms for D:\temp\repro\repro.csproj.
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.Construction.SolutionProjectGenerator.CreateTraversalInstance(String wrapperProjectToolsVersion, Boolean explicitToolsVersionSpecified, List`1 projectsInOrder)
   at Microsoft.Build.Construction.SolutionProjectGenerator.CreateSolutionProject(String wrapperProjectToolsVersion, Boolean explicitToolsVersionSpecified)
   at Microsoft.Build.Construction.SolutionProjectGenerator.Generate()
   at Microsoft.Build.Execution.ProjectInstance.GenerateSolutionWrapper(String projectFile, IDictionary`2 globalProperties, String toolsVersion, ILoggingService loggingService, BuildEventContext projectBuildEventContext, IReadOnlyCollection`1 targetNames, ISdkResolverService sdkResolverService, Int32 submissionId)
   at Microsoft.Build.Execution.ProjectInstance.LoadSolutionForBuild(String projectFile, PropertyDictionary`1 globalPropertiesInstances, String toolsVersion, BuildParameters buildParameters, ILoggingService loggingService, BuildEventContext projectBuildEventContext, Boolean isExplicitlyLoaded, IReadOnlyCollection`1 targetNames, ISdkResolverService sdkResolverService, Int32 submissionId)
   at Microsoft.Build.Execution.BuildManager.LoadSolutionIntoConfiguration(BuildRequestConfiguration config, BuildRequest request)
   at Microsoft.Build.Execution.BuildManager.HandleNewRequest(Int32 node, BuildRequestBlocker blocker)
   at Microsoft.Build.Execution.BuildManager.IssueBuildSubmissionToScheduler(BuildSubmission submission, Boolean allowMainThreadBuild)
   at Microsoft.Build.Execution.BuildManager.<>c__DisplayClass70_0.<ExecuteSubmission>b__0()
```

--- a/src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs
+++ b/src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs
@@ -2103,8 +2103,10 @@ EndGlobal
         /// <summary>
         /// Verifies that illegal user target names (the ones already used internally) don't crash the SolutionProjectGenerator
         /// </summary>
-        [Fact]
-        public void IllegalUserTargetNamesDoNotThrow()
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void IllegalUserTargetNamesDoNotThrow(bool forceCaseDifference)
         {
             SolutionFile solution = SolutionFile_Tests.ParseSolutionHelper
             (@"
@@ -2145,7 +2147,18 @@ EndGlobal
                 "ValidateProjects",
             })
             {
-                instances = SolutionProjectGenerator.Generate(solution, globalProperties, null, BuildEventContext.Invalid, CreateMockLoggingService(), builtInTargetName == null ? null : new [] { builtInTargetName });
+                string[] targetNames;
+
+                if (builtInTargetName == null)
+                {
+                    targetNames = null;
+                } else
+                {
+                    string targetName = forceCaseDifference ? builtInTargetName.ToUpperInvariant() : builtInTargetName;
+                    targetNames = new[] { targetName };
+                }
+
+                instances = SolutionProjectGenerator.Generate(solution, globalProperties, null, BuildEventContext.Invalid, CreateMockLoggingService(), targetNames);
 
                 Assert.Single(instances);
 

--- a/src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs
+++ b/src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs
@@ -2152,7 +2152,8 @@ EndGlobal
                 if (builtInTargetName == null)
                 {
                     targetNames = null;
-                } else
+                }
+                else
                 {
                     string targetName = forceCaseDifference ? builtInTargetName.ToUpperInvariant() : builtInTargetName;
                     targetNames = new[] { targetName };

--- a/src/Build/Construction/Solution/SolutionProjectGenerator.cs
+++ b/src/Build/Construction/Solution/SolutionProjectGenerator.cs
@@ -62,7 +62,7 @@ namespace Microsoft.Build.Construction
         /// <summary>
         /// A known list of target names to create.  This is for backwards compatibility.
         /// </summary>
-        internal static readonly ISet<string> _defaultTargetNames = ImmutableHashSet.Create(StringComparer.OrdinalIgnoreCase,
+        internal static readonly ImmutableHashSet<string> _defaultTargetNames = ImmutableHashSet.Create(StringComparer.OrdinalIgnoreCase,
             "Build",
             "Clean",
             "Rebuild",

